packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -90,49 +164,65 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
         }
         if (!regex.test(userInput)) {
           errorMessageToDisplay = strings.invalidInputErrorMessage;
+        } else if (timeRange) {
+          const optionDate: Date = getDateFromTimeSelection(useHour12, dateStartAnchor, userInput);
+          if (optionDate < dateStartAnchor || optionDate > dateEndAnchor) {
+            errorMessageToDisplay = format(
+              strings.timeOutOfBoundsErrorMessage,
+              dateStartAnchor.toString(),
+              dateEndAnchor,
```

Comment:
nit: also call `toString() `on `dateEndAnchor` or get rid of `toString()` on `dateStartAnchor`?

Reply:
Good catch - resolved by calling `toString()` on both Date objects.